src/zod.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  appendErrors,
+  FieldValues,
+  Resolver,
+  ResolverError,
+  ResolverSuccess,
+  transformToNestObject,
+} from 'react-hook-form';
+import { Schema, ZodError } from 'zod';
+import { ParseParams } from 'zod/lib/src/parser';
+import convertArrayToPathName from './utils/convertArrayToPathName';
+
+const parseErrorSchema = (
+  zodError: ZodError,
+  validateAllFieldCriteria: boolean,
+) => {
+  if (!Array.isArray(zodError.errors) || zodError.isEmpty) {
+    return {};
+  }
+
+  return zodError.errors.reduce<Record<string, any>>(
+    (previous, { path, message, code: type }) => {
+      const currentPath = convertArrayToPathName(path);
+
+      previous[currentPath] = {
+        message,
+        type,
+      };
```

Comment:
mutating the accumulator is usually not a good practice, this makes the reducer not pure.